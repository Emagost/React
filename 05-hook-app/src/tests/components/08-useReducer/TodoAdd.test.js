import TodoAdd from "../../../components/08-useReducer/TodoAdd";
import {shallow} from "enzyme";

describe('test <TodoAdd/>', () => {

    const handleAddTodo = jest.fn();
    const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo}/>)
    
    test('snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('shouldnt call handleAddTodo', () => {
        
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} })

        expect( handleAddTodo ).toHaveBeenCalledTimes(0);
    });

    test('should call handleAddTodo',()=>{
        wrapper.find('input').at(0).simulate('change', {
            target: {
                name: 'description',
                value: 'Learn Mongo'
            }
        });
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({preventDefault: ()=>{}});

        expect(handleAddTodo).toHaveBeenCalled();
        expect( handleAddTodo ).toHaveBeenCalledWith(expect.any(Object));
        expect(wrapper.find('input').prop('value')).toBe('')
        
        //No funciona ya que enzyme no esta adaptado para react 17
        // expect( handleAddTodo ).toHaveBeenCalledWith({
        //     id: expect.any(Number),
        //     desc: 'Learn Mongo',
        //     done: false,
        // });
    });
});
