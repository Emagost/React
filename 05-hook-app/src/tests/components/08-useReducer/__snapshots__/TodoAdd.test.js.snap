// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <TodoAdd/> snapshot 1`] = `
<Fragment>
  <h4>
    Add todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="description..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 w-100"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
